Helper/EavStructureWrapper.php

```diff
@@ -2,219 +2,124 @@
+
 class EavStructureWrapper
 {
     /**
-     * EavStructureWrapper Cache Tag
-     */
-    const SNOWDOG_MENU_DATABASE_ENTITY_ID_TAG = 'snowdog_menu_database_entity_id_tag';
-
-    /**
-     * EavStructureWrapper Cache Tag For Block
-     */
-    const SNOWDOG_MENU_DATABASE_BLOCK_ENTITY_ID_TAG = 'snowdog_menu_database_block_entity_id_tag';
-
-    /**
-     * EavStructureWrapper Cache Tag For Cms Page
-     */
-    const SNOWDOG_MENU_DATABASE_CMS_PAGE_ENTITY_ID_TAG = 'snowdog_menu_database_cms_page_entity_id_tag';
-
-    /**
-     * Enterprise Column Name
-     */
-    const ENTERPRISE_EDITION_COLUMN = 'row_id';
-
-    /**
-     * Community Column Name
-     */
-    const COMMUNITY_EDITION_COLUMN = 'entity_id';
-
-    /**
-     * Enterprise Block Column Index
-     */
-    const COMMUNITY_EDITION_BLOCK_COLUMN_INDEX = 'block_id';
-
-    /**
-     * Enterprise Cms Page Column Index
-     */
-    const COMMUNITY_EDITION_CMS_PAGE_COLUMN_INDEX = 'page_id';
-
-    /**
-     * @var ProductMetadataInterface
-     */
-    private $productMetadata;
-
-    /**
-     * @var ResourceConnection
+     * @var MetadataPool
      */
-    private $connection;
+    private $metadataPool;
 
     /**
      * @var string
      */
-    private $columnName = '';
+    private $cmsPageLinkField;
 
     /**
      * @var string
      */
-    private $blockColumnName = '';
+    private $cmsPageIdentifierField;
 
     /**
-     * @var
+     * @var string
      */
-    private $cmsPageColumnName = '';
+    private $cmsBlockLinkField;
 
     /**
-     * @var CacheInterface
+     * @var string
      */
-    private $cache;
+    private $categoryLinkField;
 
     /**
-     * EavStructureWrapper constructor.
-     *
-     * @param ProductMetadataInterface $productMetadata
-     * @param ResourceConnection $connection
-     * @param CacheInterface $cache
+     * @var string
      */
-    public function __construct(
-        ProductMetadataInterface $productMetadata,
-        ResourceConnection $connection,
-        CacheInterface $cache
-    ) {
-        $this->productMetadata = $productMetadata;
-        $this->connection = $connection;
-        $this->cache = $cache;
-        $this->initCacheVariables();
-    }
-
-    protected function initCacheVariables()
-    {
-        if ($columnName = $this->cache->load(self::SNOWDOG_MENU_DATABASE_ENTITY_ID_TAG)) {
-            $this->columnName = $columnName;
-        }
-        if ($blockColumnName = $this->cache->load(self::SNOWDOG_MENU_DATABASE_BLOCK_ENTITY_ID_TAG)) {
-            $this->blockColumnName = $blockColumnName;
-        }
-        if ($cmsPagecolumnName = $this->cache->load(self::SNOWDOG_MENU_DATABASE_CMS_PAGE_ENTITY_ID_TAG)) {
-            $this->cmsPageColumnName = $cmsPagecolumnName;
-        }
-    }
+    private $categoryIdentifierField;
 
     /**
-     * @return string
+     * @param MetadataPool $metadataPool
      */
-    public function getEdition()
+    public function __construct(MetadataPool $metadataPool)
```

Comment:
`Magento\Framework\EntityManager\MetadataPool` is available since Magento 2.1 - do we drop support for Magento 2.0 ? (also dependency on framework in at least 100.1.0 should be placed in composer.json)

If we are using `MetadataPool` I see no point of using and maintaining `EavStructureWrapper` class as it only acts as proxy for `MetadataPool` (I would use `MetadataPool` directly in resource models)

Reply:
Okay, I removed `EavStructureWrapper` and I updated composer dependencies.

Moreover, we decided to release a new minor version and drop support for old core versions.